test/tools/ossfuzz/SolidityGenerator.h

```diff
@@ -0,0 +1,42 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Implements generators for synthesizing mostly syntactically valid
+ * Solidity test programs.
+ */
+
+#pragma once
+
+#include <random>
+
+namespace solidity::test::fuzzer
+{
+using RandomEngine = std::mt19937_64;
+
+class SolidityGenerator
+{
+public:
+	SolidityGenerator(unsigned int _seed): m_rand(_seed)
+	{}
+	/// @returns a pseudo randomly generated test program
+	std::string generateTestProgram();
+private:
+	/// Random number generator
+	RandomEngine m_rand;
```

Comment:
Will `m_rand` ever change? Can it be `const`?

Reply:
`m_rand` is seeded here, so it cant be `const` iiuc

https://github.com/ethereum/solidity/pull/9722/files#diff-1b015d56d122f4307a3485f8b0c74e56df64a2da9d5cc28de179fd102d416d66R34

Essentially, each mutation has to be deterministic. To enable this, `libFuzzer` provides a unique seed for each mutation which is passed to the local `RandomEngine` instance.

Reply:
Right, but it can still be `const` and initialized in the constructor

Reply:
Hmm... If I do `RandomEngine const& m_rand` and initialize `m_rand(_seed)` the compiler complains that I am trying to bind to an lvalue of type `unsigned`. Something does not seem right

Reply:
Not a reference though, just
```suggestion
	RandomEngine const m_rand;
```